Add events for button press and release, default to not creating button Entities.

colohan opened the pull request.

colohan committed:
Rename exclude_buttons to include_buttons, change default to not include.

colohan committed:
Add events for button press and release.

gjbadros commented on __init__.py:
```diff
@@ -51,12 +52,14 @@
-                vol.Optional(CONF_EXCLUDE_VARIABLES): cv.boolean,
-                vol.Optional(CONF_INCLUDE_UNDERSCORE_VARIABLES): cv.boolean,
-                vol.Optional(CONF_DISABLE_CACHE): cv.boolean,  # FIXME
+                vol.Optional(CONF_INCLUDE_BUTTONS, default=False): cv.boolean,
```
It's a little weird that it's INCLUDE_BUTTONS but EXCLUDE_* everything else.  I think it makes sense since the default of excluding the buttons makes sense given this change, whereas we want the default to be to include for everything else.

colohan replied:
Agreed, it is weird.  I struggled over this for a bit, tried it both ways, and eventually decided that this way was marginally less evil.  But if you prefer it the other way, I'm happy to change it.  The only part I feel somewhat more strongly about is that the default should not be to include a zillion button sensors.

gjbadros commented on sensor.py:
```diff
@@ -97,10 +97,12 @@ def device_class(self):
         """Run when invoked by pyvantage when the device state changes."""
         self.schedule_update_ha_state()
 
+        if self._vantage_device.kind == "button":
```
We talked about maybe having a config directive fire_button_events, but unless there's a precedent in other modules for guarding firing of events with a config setting, they're almost certainly lightweight enough and rare enough to just let them be fired and have them be ignored.

colohan replied:
That was my gut feeling.  Happy to put a config option in if you think it matters, but otherwise I'll just leave it.

gjbadros commented on __init__.py:
```diff
@@ -77,6 +80,21 @@ def mappings_from(nm):
+def button_pressed(hass, button):
+    """Generate HASS bus events for button presses and releases."""
+    payload = {
+        'button': slugify(button.name),
```
Should we add the button number (1-5) and the keypad name and keypad_vid as fields, too? I'm trying to think of what we'd want available in a handler and I can imagine each of those things being pretty useful.

colohan replied:
Yes.  I'll add that (maybe tonight, probably tomorrow).

colohan committed:
Merge branch 'upstream' into event_button

colohan committed:
Add more useful fields to vantage_button_{pressed,released} events.

Currently directly accesses private fields in pyvantage, should fix this to use
"real" APIs.
